private/bufpkg/bufcheck/bufcheck.go

```diff
@@ -162,17 +164,35 @@ func WithPluginsEnabled() ClientFunctionOption {
 	return pluginsEnabledOption{}
 }
 
-// RunnerProvider provides pluginrpc.Runners for program names and args.
+// RunnerProvider provides pluginrpc.Runners for a given plugin config.
 type RunnerProvider interface {
-	NewRunner(programName string, programArgs ...string) pluginrpc.Runner
+	NewRunner(pluginConfig bufconfig.PluginConfig) (pluginrpc.Runner, error)
```

Comment:
I believe this is the primary motivation of this PR, otherwise, this interface is tightly coupled to executing plugins locally.

By taking a `PluginConfig,` we can in the future have constructors for both `NewLocalPluginConfig` and `NewRemotePluginConfig`. This effectively punts the `pluginConfig.Type() != bufconfig.PluginConfigTypeLocal` check to the provider. The provider now has more information about what the runner should be.